### PR TITLE
Update subaward loader to ignore dashes when attempting to identify matching awards

### PR DESCRIPTION
**Pull request by nmonga91**
**High level description:**
Update subaward loader to ignore dashes when attempting to identify matching awards

**Technical details:**
- Reorganizing helpers used across the code base
- Updating subaward loader to be "dash agnostic" 
- Added integration tests for the subaward loader. Unit tests were deemed non-essential since integration tests cover the core functionality of testing the data transformation happening with the script.
- Removed old subaward loader, which is now dead code

**Link to JIRA Ticket:**
[DEV-811](https://federal-spending-transparency.atlassian.net/browse/DEV-811)

**The following are ALL required for the PR to be merged:**
- [x] Backend review completed
- [x] Matview impact assessment completed (N/A)
- [x] Frontend impact assessment completed (N/A)
- [ ] Data validation completed
  - Unit tests show that the subawards are mapping correctly. Actual subaward loads will need to occur to truly verify a fix. That will be a pending data check to happen on staging with real data.
- [x] API Performance evaluation completed and present (N/A)

### CODE REVIEW

File: usaspending_api/settings.py [hunk 1]
```diff
@@ -247,7 +247,7 @@
     },
 }
 
-# If caches added or renamed, edit clear_caches in usaspending_api/etl/helpers.py
+# If caches added or renamed, edit clear_caches in usaspending_api/etl/generic_helper.py
```
Comment (tony-sappe) [inline usaspending_api/settings.py]: Trigger-happy find-replace?

File: usaspending_api/broker/management/commands/load_fsrs.py [hunk 2]
```diff
@@ -49,7 +49,23 @@ def get_award_data(db_cursor, award_type, max_id, ids=None):
...
+                'CONT_AW_' + \
+                (row['contract_agency_code'].replace('-', '') if row['contract_agency_code'] else '-NONE-') + \
+                '_' + \
+                (row['contract_idv_agency_code'].replace('-', '') if row['contract_idv_agency_code'] else '-NONE-') + \
+                '_' + \
+                (row['contract_number'].replace('-', '') if row['contract_number'] else '-NONE-') + \
+                '_' + \
+                (row['idv_reference_number'].replace('-', '') if row['idv_reference_number'] else '-NONE-')
+        else:
+            return row['fain'].replace('-', '')
```
Comment (tony-sappe) [inline usaspending_api/broker/management/commands/load_fsrs.py]: Is this correct? Doesn't appear to be creating a `CONT_AW_...` string

File: usaspending_api/broker/tests/data/broker_subawards.json [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    {
+      "internal_id": "PROCUREMENT_INTERNAL_ID",
+      "id": "-1",
+      "report_period_mon": "11",
+      "report_period_year": "1111",
+      "duns": "DUNS12345",
+      "parent_duns": "PARENTDUNS12345",
+      "principle_place_country": "USA",
+      "principle_place_city": "Pawnee",
```
Comment (tony-sappe) [inline usaspending_api/broker/tests/data/broker_subawards.json]: http://i0.kym-cdn.com/photos/images/original/000/611/135/fca.png

File: usaspending_api/conftest.py [hunk 4]
```diff
@@ -1,5 +1,80 @@
...
+    """Mocks all matvies to a single mock queryset"""
+    mock_qs = MockSet()  # mock queryset
+
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryPscCodesView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryNaicsCodesView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryCfdaNumbersView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryTransactionMonthView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.SummaryTransactionView.objects', mock_qs)
+    monkeypatch.setattr('usaspending_api.awards.models_matviews.UniversalTransactionView.objects', mock_qs)
```
Comment (tony-sappe) [inline usaspending_api/conftest.py]: Missing three materialized views

Comment (nmonga91) [reply usaspending_api/conftest.py]: This was added from a separate branch which was only working off transactions at the time (the branch for spending by category). I've updated it to include the missing 3 regardless.

File: usaspending_api/broker/management/commands/load_fsrs.py (hunk 2 above)
Comment (nmonga91) [reply usaspending_api/broker/management/commands/load_fsrs.py]: Added comments. Assistance award identifiers typically are prepended with `ASST_AW_` but we don't have all the necessary fields to actually generate it, so we can only go off of the `fain`.

**Comment by nmonga91** [comment]
@tony-sappe the bug you mentioned should be resolved. A fix was pushed into dev and that has been down-merged into this branch. All other comments have been addressed.

**Pull request merged**